ff/src/fields/mod.rs

```diff
@@ -42,7 +44,49 @@ use ark_std::cmp::max;
 #[cfg(feature = "parallel")]
 use rayon::prelude::*;
 
-/// The interface for a generic field.  
+
+pub trait AdditiveGroup:
+    Eq
+    + 'static
+    + Sized
+    + CanonicalSerialize
+    + CanonicalDeserialize
+    + Copy
+    + Clone
+    + Default
+    + Send
+    + Sync
+    + Hash
+    + Debug
+    + Display
+    + UniformRand
+    + Zeroize
+    + Zero
+    + Neg<Output = Self>
+    + Add<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + Mul<<Self as AdditiveGroup>::ScalarField, Output = Self>
+    + AddAssign<Self>
+    + SubAssign<Self>
+    + MulAssign<<Self as AdditiveGroup>::ScalarField>
+    + for<'a> Add<&'a Self, Output = Self>
+    + for<'a> Sub<&'a Self, Output = Self>
+    + for<'a> Mul<&'a <Self as AdditiveGroup>::ScalarField, Output = Self>
+    + for<'a> AddAssign<&'a Self>
+    + for<'a> SubAssign<&'a Self>
+    + for<'a> MulAssign<&'a <Self as AdditiveGroup>::ScalarField>
+    + for<'a> Add<&'a mut Self, Output = Self>
+    + for<'a> Sub<&'a mut Self, Output = Self>
+    + for<'a> Mul<&'a mut <Self as AdditiveGroup>::ScalarField, Output = Self>
+    + for<'a> AddAssign<&'a mut Self>
+    + for<'a> SubAssign<&'a mut Self>
+    + for<'a> MulAssign<&'a mut <Self as AdditiveGroup>::ScalarField>
+    + core::iter::Sum<Self>
+    + for<'a> core::iter::Sum<&'a Self> {
+        type ScalarField: Field;
```

Comment:
Does it suffice to only specify the `ScalarField` associated type? I think we should move also some other associated items here:
* `additive_generator()`/`one()`
* `double_in_place` and `double`
* `neg_in_place`
* `ZERO`

Reply:
zero is already there -- but one could also not exist in an additive group! e.g. {0}.
As for doubling and inverse formulae, that's a good point, but I'd like to move them to this trait in a separate pr. I can create a new issue for this.

Reply:
by `ZERO` I meant the `const ZERO: Self` that exists currently. Does moving `double`, `neg`, require adding new imports in more places?

Reply:
already with `const ZERO` it would. I added a commitment for it. Should I proceed with the other ones?

Reply:
Yeah, go ahead. I think it's a good idea to minimize the code churn to one PR

Reply:
aye aye, done!